src/chexus/validate.py

```diff
@@ -64,7 +67,13 @@ def validate(group: Group, validators: list[Validator]) -> dict[type, Validation
     results = {type(v): ValidationResult(v) for v in validators}
     for node in tree.values():
         for validation in results.values():
-            validation.apply(node)
+            try:
+                validation.apply(node)
+            except ModuleNotFoundError as e:
+                if e.name != 'scipp':
+                    raise e
+                validation.skipped = True
+
```

Comment:
Instead of this, maybe add an argument to `base_validators` that deselecteds scipp-based validators if the flag is set.

Reply:
I thought about that but decided against it for the reasons mentioned above. I'm open to doing it that way but I'm curious to know why you think that's better?

Reply:
Many people (maybe fewer in Python than in other languages) feel that doing "regular" control flow via exceptions is an anti-pattern (it should be "exceptions", as the name implies). In this case you can well encounter this try-except 10s or 100s of thousands of times for any given file, which is also not so nice.

Reply:
I agree that it's an anti pattern if there's another canonical way to do the control flow. (The classical example would be trying to get a value from a dictionary that maybe doesn't exist, and using try/except instead of `get`). However, for control flow regarding non-avaliable imports I'd say try except doesn't really have an equivalent canonical way.

Regarding performance I think that's unlikely to be an issue. My understanding is that try-except only has performance implications if the exception is triggered. That only happens when the scipp based validators are applied to nodes where they are applicable and that's unlikely to be a large fraction of the total.

Reply:
Note that your current approach will also break if someone writes a validator that imports in one of the other methods.

Reply:
I agree that using excepts to handle import failures is idiomatic. But then it should be tightly placed around the import. Here, you can't tell which import raised the exception.
So how about checking whether scipp is available and whether the command line argument is set and then setting a config option somewhere? (I'm not familiar enough with the code to say where that would go.) And then the validators that use scipp would check that flag and not even run if the check is disabled.